app/src/main/java/org/xbmc/kore/ui/widgets/MediaProgressIndicator.java

```diff
@@ -0,0 +1,143 @@
+        super(context, attributeSet);
+        initializeView(context);
+    }
+
+    private void initializeView(Context context) {
+        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        inflater.inflate(R.layout.media_progress_indicator, this);
+    }
+
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+
+        seekBar = (SeekBar) findViewById(R.id.mpi_seek_bar);
+        progressTextView = (TextView) findViewById(R.id.mpi_progress);
+        durationTextView = (TextView) findViewById(R.id.mpi_duration);
+
+        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                if (fromUser) {
+                    MediaProgressIndicator.this.progress = progress;
+
+                    progressTextView.setText(UIUtils.formatTime(MediaProgressIndicator.this.progress));
+                }
+            }
+
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+                // Stop the seekbar from updating
+                seekBar.removeCallbacks(seekBarUpdater);
+            }
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+                if (onProgressChangeListener != null)
+                    onProgressChangeListener.onProgressChanged(seekBar.getProgress());
+
+                if (speed > 0)
+                    seekBar.postDelayed(seekBarUpdater, SEEK_BAR_UPDATE_INTERVAL);
+            }
+        });
+    }
+
+    private Runnable seekBarUpdater = new Runnable() {
+        @Override
+        public void run() {
+            if ((maxProgress == 0) || (progress >= maxProgress)) {
+                seekBar.removeCallbacks(this);
+                return;
+            }
+
+            progress += progressIncrement;
+            setProgress(progress);
+
+            seekBar.postDelayed(this, SEEK_BAR_UPDATE_INTERVAL);
+        }
+    };
+
+    public void setOnProgressChangeListener(OnProgressChangeListener onProgressChangeListener) {
+        this.onProgressChangeListener = onProgressChangeListener;
+    }
+
+    public void setProgress(int progress) {
+        this.progress = progress;
+        seekBar.setProgress(progress);
+        progressTextView.setText(UIUtils.formatTime(progress));
+    }
+
+    public void setMaxProgress(int max) {
+        maxProgress = max;
+        seekBar.setMax(max);
+        durationTextView.setText(UIUtils.formatTime(max));
+    }
+
+    /**
+     * Sets the play speed for the progress indicator
+     * @param speed set to 0 to stop updating the progress indicator.
+     */
+    public void setSpeed(int speed) {
+        if( speed == this.speed )
+            return;
+
+        this.speed = speed;
+        this.progressIncrement = speed * (SEEK_BAR_UPDATE_INTERVAL/1000);
```

Comment:
I'm using speed here to allow for different speeds. Although the original implementation only supported speed 1 I can't think of a reason why it can't be different. Is there a problem with speeds greater or smaller (besides 0) than 1?